test/src/unit-cppapi-profile.cc

```diff
@@ -177,6 +177,60 @@ TEST_CASE_METHOD(
   }
 }
 
+TEST_CASE_METHOD(
+    ProfileCPPFx,
+    "C++ API: Profile load validation",
+    "[cppapi][profile][load]") {
```

Comment:
Very useful tests! Can you add some variations where you use the defaults as well so that we confirm they work as expected? I.e. a) call the default constructor `Profile()` , b) call the same constructor you are calling but providing all 3 combinations of `std::nullopt` : `(std::nullopt, std::nullopt) , (name_, std::nullopt) , (std::nullopt, tempdir_.path()` ? 
What 'd be more interested in these cases is to check if the files were created were we expect in the filesystem, and perhaps what happens if you try to save twice using the default constructor.